### ISSUE TITLE
[E2E Tests][CI] Run the e2e tests on the Release version of the app

### ISSUE THREAD

**Issue by alexjba** [status-im/status-desktop#13079]
### Description
The purpose of this task is to explore the possibility of running the e2e tests on the release version of the app.
The release version can be compiled by using the RELEASE env variable. This is configured in the Jenkins files to false by default and currently there is no way to set the release flag on e2e pipeline.

IMO there are 2 checkpoints for this task:
1. Try the e2e tests on Release builds
2. Set-up the CI to run the squish tests on Release builds.

The motivation for this task is to reduce the CI time and to test the app version available to users.

**Comment by anastasiyaig** [comment]
@jakubgs @apentori can you please take a look?

**Comment by jakubgs** [comment]
Do you simple mean to enable `RELEASE` flag for `nightly` builds?
https://ci.status.im/job/status-desktop/job/nightly/

**Comment by alexjba** [comment]
@jakubgs I was referring to this job https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-e2e/.

**Comment by jakubgs** [comment]
What? You want to create __yet another__ build that is a release build for each and every single PR?

This is a getting a bit silly. I'm not going to double the number of builds for all PRs. We can make current PR builds use `RELEASE=true` if that's what you want.

**Comment by alexjba** [comment]
> What? You want to create **yet another** build that is a release build for each and every single PR?
> 
> This is a getting a bit silly. I'm not going to double the number of builds for all PRs. We can make current PR builds use `RELEASE=true` if that's what you want.

No, I definitely don't want another build to run on each PR. The goal is to speed things up 😄 AFAIK this job builds the app and then runs squish on it. Running squish on an app built with RELEASE flag would speed things up.

**Comment by jakubgs** [comment]
Oh, I see, you're referring to the __OLD__ E2E tests, which build the app as part of the run:
https://github.com/status-im/status-desktop/blob/3eff96bfc84a9833f0b27022b83085a9c6b81081/ci/Jenkinsfile.tests-e2e#L95-L99

Instead of the __NEW__ E2E tests which simply reuse the app already build for the PR:
```goovy
        copyArtifacts(
          projectName: params.BUILD_SOURCE,
          filter:      'pkg/*-x86_64.tar.gz',
          selector:    lastWithArtifacts(),
          target:      './'
        )
```
https://github.com/status-im/desktop-qa-automation/blob/cc0b08924ef7984efc2659a31a01faab4bddab9f/ci/Jenkinsfile#L116-L121

My question would be... why are we still running the __OLD__ E2E tests? @anastasiyaig ?

**Comment by jakubgs** [comment]
Aside from why are we running old E2E still, if we want them to use `RELEASE=true` we can just add it in `environment` secion:
https://github.com/status-im/status-desktop/blob/3eff96bfc84a9833f0b27022b83085a9c6b81081/ci/Jenkinsfile.tests-e2e#L63-L79

Or alternatively as a parameter, if we want to control it at runtime.

**Comment by anastasiyaig** [comment]
> My question would be... why are we still running the OLD E2E tests? @anastasiyaig ?

finally we dropped them, it took really long , sorry about that.

For new tests, i just need to add env var as you suggests or there is something else @jakubgs ?

**Comment by jakubgs** [comment]
Since new/current E2E tests reuse the application built as part of the PR and don't build one themselves the change needs to be done in the Jenkinsfile responsible for the Linux PR build:
```groovy
    booleanParam(
      name: 'RELEASE',
      description: 'Decides whether binaries are built with debug symbols.',
      defaultValue: params.RELEASE ?: false
    )
```
https://github.com/status-im/status-desktop/blob/94f250d67173c89f36f3ef7074bd8f37cd90f3e1/ci/Jenkinsfile.linux#L19-L23

As we can see here the builds by default use `RELEASE=false`. We can either make this default to `true`, or write some kind of conditional function, if we want to have different defaults for different cases. But it would probably make most sense just to make this `true`.

**Comment by anastasiyaig** [comment]
thanks @jakubgs all clear

**Comment by anastasiyaig** [comment]
so i opened a simple draft PR to test this and the app is not building because of token https://github.com/status-im/status-desktop/pull/13675

```
[2024-02-22T08:14:54.410Z] Using POKT token: release
Could not find credentials entry with ID 'infura-api-release'
```

I suspect `release` is not just managing debug symbols presence. Can we build without debug symbols but with DEV tokens @jakubgs ?

**Comment by jakubgs** [comment]
Not according to this conditional:
https://github.com/status-im/status-desktop/blob/94f250d67173c89f36f3ef7074bd8f37cd90f3e1/Makefile#L212-L220
You would have to modify it, complicate it in some way. Add a secondary condition or something like that.

**Comment by anastasiyaig** [comment]
@alexjba do you know the possible solution?

**Comment by alexjba** [comment]
> ```
> [2024-02-22T08:14:54.410Z] Using POKT token: release
> Could not find credentials entry with ID 'infura-api-release'
> ```


> so i opened a simple draft PR to test this and the app is not building because of token #13675
> 
> ```
> [2024-02-22T08:14:54.410Z] Using POKT token: release
> Could not find credentials entry with ID 'infura-api-release'
> ```
> 
> I suspect `release` is not just managing debug symbols presence. Can we build without debug symbols but with DEV tokens @jakubgs ?

I don't see any connection between the debug symbols and this error. I think we only need to set-up the POKT/INFURA keys. Not sure how the setup looks like on release.
@alaibe Do you know how can we fix this? We see this error in the e2e builds if we're using the release version of the app.

**Comment by jakubgs** [comment]
The connection is very simple.

- Develop CI(https://ci.status.im/) does not hold release credentials.
- Release CI(https://ci.infra.status.im/) has release credentials.

Those two shall not mix. Hence a real release build cannot be built on devel CI.
Also, it shouldn't, since PR builds should not be release builds, signed with release certificates, and contain release credentials.

**Comment by jakubgs** [comment]
We could remove the `params.RELEASE == true` from the `isReleaseBuild()` conditional, and that would prevent the PR/develop build from using release credentials, while also not containing debug symbols, but that would also confuse the meaning of `RELEASE` flag. If that's what we want to do, and this is the only purpose of the `RELEASE` flag, then maybe we should just rename it and invert it, so it's called `DEBUG_BUILD` or `INCLUDE_DEBUG_SYMBOLS` for clarity.

**Comment by anastasiyaig** [comment]
> rename it and invert it, so it's called DEBUG_BUILD or INCLUDE_DEBUG_SYMBOLS for clarity.

yes, we need to handle option of having / not having debug symbols separately, indeed

**Comment by jakubgs** [comment]
Okay, then that seems to be the best option here. Unless there is some other use of `RELEASE` env variable that I'm not aware of that will break if we remove it.

**Comment by jakubgs** [comment]
Is there any point in keeping the `RELEASE` parameter tho? Or can we just replace both the env variable and the parameter with `INCLUDE_DEBUG_SYMBOLS`?

**Comment by anastasiyaig** [comment]
lets ask @jrainville maybe

**Comment by jakubgs** [comment]
There is one potential use-case I can see, which is that the meta-job(combined) provides the `RELEASE` flag to the child jobs:
```groovy
/* Start child job with some default and custom parameters. */
def Build(String name, Map extraParams=[:]) {
  def defaultParams = [
    /* necessary for updating GitHub PRs */
    'CHANGE_ID':  env.CHANGE_ID,
    'BUILD_TYPE': utils.getBuildType(),
    'RELEASE':    utils.isReleaseBuild(),
```
https://github.com/status-im/status-jenkins-lib/blob/daecfe7d808daac139d4e4a3606378d9db0746c5/vars/jenkins.groovy#L63

This is the only utility I see for it.

**Comment by jakubgs** [comment]
Here's an attempt:

* https://github.com/status-im/status-desktop/pull/13701

**Comment by alexjba** [comment]
> We could remove the `params.RELEASE == true` from the `isReleaseBuild()` conditional, and that would prevent the PR/develop build from using release credentials, while also not containing debug symbols, but that would also confuse the meaning of `RELEASE` flag. If that's what we want to do, and this is the only purpose of the `RELEASE` flag, then maybe we should just rename it and invert it, so it's called `DEBUG_BUILD` or `INCLUDE_DEBUG_SYMBOLS` for clarity.

I think it's getting confusing because the concepts are kind of mixed together, mixing the `how` it's built with `what` is included in the app. `INCLUDE_DEBUG_SYMBOLS` could be enough for now, but we could bring in more clarity if you think it's worth.

IMO we could have the `build type` to describe how the app is built. It could either be `RELEASE` or `DEBUG`. It will describe the optimisation level and the debug symbols level (BTW, `INCLUDE_DEBUG_SYMBOLS=false` will still include some symbols).
And something like `flavor` to describe what's included in the app. It could be `dev` - nothing extra included, `prod` - all the release keys, `test` - it's a custom prod version to include all the testing features we have. As a result we would be free to choose from all the different configurations: devDebug, devRelease, testDebug, testRelease, prodDebug, prodRelease. Each configuration would serve a specific purpose.

Do you think this will help?

**Comment by anastasiyaig** [comment]
https://github.com/status-im/status-desktop/pull/13701 is merged already that should address this task
